### ISSUE TITLE
coco_interest_class.txt

### ISSUE THREAD

**Issue by ChunmianLin** [nicholaskajoh/Vehicle-Counting#23]
hi, thanks for your great job. When I used tfoda for vehicle detection and counting, I encounted some problems that maybe need your help. I want to how to get the file 'coco.classes.txt' and 'coco_interest.class.txt'.
Thanks a lot in advance.

**Comment by nicholaskajoh** [comment]
coco.classes.txt
```
person
bicycle
car
motorcycle
airplane
bus
train
truck
boat
traffic light
fire hydrant
stop sign
parking meter
bench
bird
cat
dog
horse
sheep
cow
elephant
bear
zebra
giraffe
backpack
umbrella
handbag
tie
suitcase
frisbee
skis
snowboard
sports ball
kite
baseball bat
baseball glove
skateboard
surfboard
tennis racket
bottle
wine glass
cup
fork
knife
spoon
bowl
banana
apple
sandwich
orange
broccoli
carrot
hot dog
pizza
donut
cake
chair
couch
potted plant
bed
dining table
toilet
tv
laptop
mouse
remote
keyboard
cell phone
microwave
oven
toaster
sink
refrigerator
book
clock
vase
scissors
teddy bear
hair drier
toothbrush
```

coco_interest.class.txt
```
bicycle
car
motorcycle
bus
truck
```

**Comment by nicholaskajoh** [comment]
coco.classes.txt
```
person
bicycle
car
motorcycle
airplane
bus
train
truck
boat
traffic light
fire hydrant
stop sign
parking meter
bench
bird
cat
dog
horse
sheep
cow
elephant
bear
zebra
giraffe
backpack
umbrella
handbag
tie
suitcase
frisbee
skis
snowboard
sports ball
kite
baseball bat
baseball glove
skateboard
surfboard
tennis racket
bottle
wine glass
cup
fork
knife
spoon
bowl
banana
apple
sandwich
orange
broccoli
carrot
hot dog
pizza
donut
cake
chair
couch
potted plant
bed
dining table
toilet
tv
laptop
mouse
remote
keyboard
cell phone
microwave
oven
toaster
sink
refrigerator
book
clock
vase
scissors
teddy bear
hair drier
toothbrush
```

coco_interest.classes.txt
```
bicycle
car
motorcycle
bus
truck
```

**Comment by ChunmianLin** [comment]
@nicholaskajoh Thanks!

**Comment by ChunmianLin** [comment]
By the way, can I understand 'coco_interest.classes.txt' that contains classes of interest for specific task?

**Comment by nicholaskajoh** [comment]
coco_interest.classes.txt contains the list of objects you're interested in detecting. If you want to identify all the objects your detector can detect then `CLASSES_PATH` and `CLASSES_OF_INTEREST_PATH` would point to the same file/list.

**Comment by ChunmianLin** [comment]
Thanks. When I used mask rcnn model for vehicle detection, there is no any box result presented in the resulting video. What is the problem and how can I do to get desired detection result?  Also, I have try to use faster rcnn model to detect vehicle, but it seems poor performance than that of yolo model. It is unreasonable? In terms of performance, faster rcnn should have been much better than yolo. Therefore, could you please give me some help like parameter setting or some certain operations for better performance on vehicle?  
Thanks a lot in advance!




            
            
                
                    
                            
                                
                            
                            
                                cmlin
                            
                    
                        
                            
                                    邮箱：cmlin@buaa.edu.cn
                            
                        
                
            
        Signature is customized by Netease Mail Master


On 08/17/2019 22:58, Nicholas Kajoh wrote: coco_interest.classes.txt contains the list of objects you're interested in detecting. If you want to identify all the objects your detector can detect then CLASSES_PATH and CLASSES_OF_INTEREST_PATH would point to the same file/list.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/nicholaskajoh/Vehicle-Counting/issues/23?email_source=notifications\u0026email_token=AMWSBR25QOPNIF2SEDI3UJDQFAGXPA5CNFSM4IMHHQ7KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4QNFTQ#issuecomment-522244814",
"url": "https://github.com/nicholaskajoh/Vehicle-Counting/issues/23?email_source=notifications\u0026email_token=AMWSBR25QOPNIF2SEDI3UJDQFAGXPA5CNFSM4IMHHQ7KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4QNFTQ#issuecomment-522244814",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by nicholaskajoh** [comment]
I used Faster RCNN Inception v2 trained on COCO and the results were quite good. Although YOLO has generally performed better on my test data.
If these pre-trained models are not doing well on your data, you might want to consider training your own custom model.

**Comment by ChunmianLin** [comment]
Thanks. The results of yolo, faster rcnn and mask rcnn are as follows from top to bottom, respectively. You know, the result of yolo is relatively best. I don't know where is the problem that leads to it: many false positive bbox predictions are generated when I used faster rcnn model, and there is no any bbox presented when I used mask rcnn. Need I do some fune-tuning or modifications, but I don't know how to start? In your opinion, what can I do next?
![yolo](https://user-images.githubusercontent.com/53289159/63218902-e0aaf200-c198-11e9-8f26-506b358cf67e.png)
![faster rcnn](https://user-images.githubusercontent.com/53289159/63218905-e4d70f80-c198-11e9-94a3-c2632d0da854.png)
![mask rcnn](https://user-images.githubusercontent.com/53289159/63218907-e6083c80-c198-11e9-85f1-70f0799bb42a.png)
Also, as you mentioned above, what is my custom model you mean? You know, in the area of object detection, rcnn model and yolo model are mainstream option. It is not possible that I design a new model with regards to vehicle detection application in a short time. So, I have to try to experiment these model by transfer learning, and expect to get desired result as good as possible.
Thanks for your kindness, and I am looking forward to your constructive advice.

**Comment by nicholaskajoh** [comment]
I don't mean you should create your own neural net architecture like YOLO or RCNN. I mean you should train a YOLO or RCNN model with custom data which is similar to the objects what you want to detect so as to obtain more accurate results.
In terms of getting the best out of these pretrained models, I recommend you record your video at a high enough frame rate (30 FPS and up). Vehicles driving along the road are usually travelling at high speeds hence low frame rate captures tend to contain a lot of blurred images. You might not notice it when watching the video because your eyes/brain smoothen(s) it out. Blurred images can be problematic for object detectors. If you capture at a high frame rate, you can avoid this to a reasonable degree.

**Comment by ChunmianLin** [comment]
Thanks a lot. But I still have some confusions that maybe you can help me.

1. I want to know how to calculate the AP or mAP value in your code so that I could evaluate the performance of my model. In orther word, where does the code of AP calculation locate?

2. I don't find 'log' file when I run out my own data and want to check the detection result.

3. I want to know how to count the number of vehicle in your code, how can I find the process of calculation in your code. Also, I find that the counting number in the left-top of resulting video cannot match to the top number of predicted bbox of vehicle when it presents in the frame immediately. So how can I deal with thie problems.

So sorry for my such many problems due to I am a beginner. I am grateful for your help and kindness, and am looking forward to your letter. Thanks in advance.

Sincerely.




Chunmian Lin


-----原始邮件-----
发件人:"Nicholas Kajoh" <notifications@github.com>
发送时间:2019-08-18 21:48:52 (星期日)
收件人: nicholaskajoh/Vehicle-Counting <Vehicle-Counting@noreply.github.com>
抄送: ChunmianLin <cmlin@buaa.edu.cn>, Author <author@noreply.github.com>
主题: Re: [nicholaskajoh/Vehicle-Counting] coco_interest_class.txt (#23)



I don't mean you should create your own neural net architecture like YOLO or RCNN. I mean you should train a YOLO or RCNN model with custom data which is similar to the objects what you want to detect so as to obtain more accurate results.
In terms of getting the best out of these pretrained models, I recommend you record your video at a high enough frame rate (30 FPS and up). Vehicles driving along the road are usually travelling at high speeds hence low frame rate captures tend to contain a lot of blurred images. You might not notice it when watching the video because your eyes/brain smoothen(s) it out. Blurred images can be problematic for object detectors. If you capture at a high frame rate, you can avoid this to a reasonable degree.

**Comment by nicholaskajoh** [comment]
1. AP or mAP are concerned with object detection models and not the vehicle counting system (VCS) directly. There are different ways to obtain these metrics depending on the object detection model/framework you're using.

2. If you run the VCS with the record argument (`--record`), a log file (_log.txt_) is created in the root folder of the project. There's ongoing work to overhaul logging so this will change very soon.

3. After instantiating the VCS (`vehicle_counter = VehicleCounter(*params)`), you can call the `get_count()` method to get the current vehicle count. Counting can be done in two ways: a) via a counting line (i.e when a vehicle crosses the line, it's counted) and b) without a counting line (i.e when a vehicle exits the frame, it's counted).

**Comment by ChunmianLin** [comment]
Thanks.  I will check  these problems next.




            
            
                
                    
                            
                                
                            
                            
                                cmlin
                            
                    
                        
                            
                                    邮箱：cmlin@buaa.edu.cn
                            
                        
                
            
        Signature is customized by Netease Mail Master


On 08/20/2019 01:22, Nicholas Kajoh wrote: 

AP or mAP are concerned with object detection models and not the vehicle counting system (VCS) directly. There are different ways to obtain these metrics depending on the object detection model/framework you're using.


If you run the VCS with the record argument (--record), a log file (log.txt) is created in the root folder of the project. There's ongoing work to overhaul logging so this will change very soon.


After instantiating the VCS (vehicle_counter = VehicleCounter(*params)), you can call the get_count() method to get the current vehicle count. Counting can be done in two ways: a) via a counting line (i.e when a vehicle crosses the line, it's counted) and b) without a counting line (i.e when a vehicle exits the frame, it's counted).



—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/nicholaskajoh/Vehicle-Counting/issues/23?email_source=notifications\u0026email_token=AMWSBRZWWSPHEJ3KMW75ZWTQFLJERA5CNFSM4IMHHQ7KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4TV4WA#issuecomment-522673752",
"url": "https://github.com/nicholaskajoh/Vehicle-Counting/issues/23?email_source=notifications\u0026email_token=AMWSBRZWWSPHEJ3KMW75ZWTQFLJERA5CNFSM4IMHHQ7KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4TV4WA#issuecomment-522673752",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
